external/storm-druid/src/main/java/org/apache/storm/druid/trident/DruidBeamState.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.storm.druid.trident;
+
+import com.metamx.tranquility.beam.Beam;
+import com.twitter.util.Await;
+import org.apache.storm.trident.state.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.collection.JavaConversions;
+
+import java.util.List;
+
+
+/**
+ * Trident {@link State} implementation for Druid.
+ */
+public class DruidBeamState<E> implements State  {
+    private static final Logger LOG = LoggerFactory.getLogger(DruidBeamState.class);
+
+    Beam<E> beam = null;
+
+    public DruidBeamState(Beam<E> beam) {
+        this.beam = beam;
+    }
+
+    public void send(List<E> events ) {
+        try {
+            LOG.info("Sending %d events", events.size());
+            Await.result(beam.sendBatch(JavaConversions.collectionAsScalaIterable(events).toList()));
+        } catch (Exception e) {
+            final String errorMsg = "Failed in writing messages to Druid";
+            LOG.error(errorMsg, e);
+            throw new RuntimeException(errorMsg);
+        }
+    }
+
+    public void close() {
+        try {
+            Await.result(beam.close());
+        } catch (Exception e) {
+            final String errorMsg = "Error while closing Druid beam client";
+            LOG.error(errorMsg, e);
+            throw new RuntimeException(errorMsg);
+        }
+    }
+
+    @Override
+    public void beginCommit(Long txid) {
```

Comment:
@harshach most of the trident states ignore the `beginCommit` and `commit`, because it may be very difficult to handle idempotent updates in a generic state. beginCommit and commit is currently implemented only in HdfsState. The downside is that such states would provide only at-least once semantics.

However key value based stores can provide an `IBackingMap` implementation whereby exactly once semantics can be supported. For e.g. the Redis trident implementation provides a RedisMapState. 

If druid supports key/value based abstraction, the connector should probably provide a MapState implementation. If not the beginCommit and commit should be handled within the DruidBeamState if we are looking to provide exactly once.